### ISSUE TITLE
"Could not create SSL/TLS secure channel" on Win server 2008 R2

### ISSUE THREAD

**Issue by Asthi** [minio/minio#5834]
After restricting cipher suites there is a problem with connecting to minio server from Windows Server 2008 R2 and Windows 7 (they are old but not dead yet) with simple WebRequest from .NET client or even Powershell.
This "security update" is forcing me to disable SSL what is unacceptable.

Supported cipher suites before updating our Minio server:
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256  (0xC02F)
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384  (0xC030) 
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA  (0xC013) 
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA  (0xC014) 
TLS_RSA_WITH_AES_128_GCM_SHA256  (0x9C) 
TLS_RSA_WITH_AES_256_GCM_SHA384  (0x9D)   
TLS_RSA_WITH_AES_128_CBC_SHA  (0x2F) 
TLS_RSA_WITH_AES_256_CBC_SHA  (0x35) 
TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA  (0xC012)
TLS_RSA_WITH_3DES_EDE_CBC_SHA  (0xA)

And after updating:
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256  (0xC02F)
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384  (0xC030) 
(Both not supported by Windows Server 2008 R2 - [TLS Cipher Suites in Windows 7](https://msdn.microsoft.com/en-us/library/windows/desktop/mt767780(v=vs.85).aspx)

## Possible Solution
Please give administrators the ability to decide on the level of security or at least restore the support of the *_CBC_SHA cipher suites.

**Comment by aead** [comment]
Thanks for opening @Asthi  - I'm aware of the Windows 7/8 TLS issue. You are using a RSA private/public key pair for your certificate. I recommend that you switch to an ECDSA based certificate. The cipher suite `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P256` is supported by both Windows 7 and the minio S3 server. You can generate an ECDSA key pair by following this [doc](https://docs.minio.io/docs/how-to-secure-access-to-minio-server-with-tls).

TL;DR: We disabled all currently unsupported cipher-suites on purpose and it's possible (AFAIK) to
use many (also Win7/8) clients with the current TLS configuration. 

Let me explain our reasons for disabling certain cipher suites.
 - We disabled `*_CBC_SHA256` and `*_CBC_SHA384` because the current Go TLS stack does not contain effective countermeasures against CBC timing attacks against both cipher suite classes.
- We disabled `*CBC_SHA` because SHA-1 is a broken cryptographic hash function. Even though using it as a MAC - like TLS - is still fine at the moment, it should not be used anymore.

So the workaround would be to use a ECDSA certificate.

**Comment by deekoder** [comment]
@Asthi  Please let us know if you want to discuss this further and think it's not addressed.
